packages/block-editor/src/components/block-list/block-popover.js

```diff
@@ -115,9 +115,17 @@ function BlockPopover( {
 	let anchorRef = node;
 
 	if ( hasMultiSelection ) {
+		const bottomNode = blockNodes[ lastClientId ];
+
+		// Wait to render the popover until the bottom reference is available
+		// as well.
```

Comment:
Is there something that triggers a re-render when the reference become available. What guarantee us that the rerender will happen?

Reply:
They are available through state provided by context. When there is a multi selection, the first and the last multi selected blocks both set their reference in this state, also creating a new array, so it causes a re-render for this component. The first and last node are not set at the same time though, so we have to wait until both are set.